### DISCUSSION TITLE
ioGetOutput Functions

### DISCUSSION THREAD

**Discussion by AshleyJamesy** [wiremod/wire#2776] [Suggestions]
Currently you can only ioSetOutput, but what if you want to get an output value programatically by name?

Outputs can be set and read so it makes sense they should also have a ioGetOutput function.
  
Example:  
```
for(I = 1, 10) {
    print(ioGetOutputNumber("Output" + I))
}
```

Currently the only way to achieve something similar is to use a wirelink:  
```
local Wirelink = wirelink()
print(Wirelink["someoutput", number])
```

But this is not readily apparent to some coders and kind of hacky

**Comment by vurvdev** [comment]
Why would you want to get the value of an output when you're the one setting it?

**Reply by AshleyJamesy** [reply]
To **programatically** check the output before deciding what to do next. eg:  
Probably not the best example, but this is an actual thing I've tried to do for fading doors so I don't have a big switch or elseif statement.
(only provided relevant code, not entire code)

```
@outputs Fade1 Fade2 Fade3 Fade4, ... etc

if(clkName():startsWith("close-")) {
    local N = clkName():sub(7, clkName():length())
    if(ioGetOutputNumber("Fade" + N) == 1)  {
        #do other stuff if fade is open
        ioSetOutputNumber("Fade" + N, 0) #close the fading door
    } else {
        #cannot close, door is not open, do something here
    }
}
```

but instead I need to do this  
  
```
@outputs Fade1 Fade2 Fade3 Fade4, ... etc

if(clkName():startsWith("close-")) {
    local N = clkName():sub(7, clkName():length())

    local Wirelink = wirelink()
    if(Wirelink["Fade" + N, number] == 1) {
        #do other stuff if fade is open
        ioSetOutputNumber("Fade" + N, 0) #close the fading door
    } else {
        #cannot close, door is not open, do something here
    }
}
```

It's possible to read outputs right now, but ideally since you can set and read outputs like `Output = 1, print(Output)` it makes sense that we should also be able to read outputs using a `ioGetOutput` - I hope I'm making sense.

`ioSetOutput` is the equivalent of `Output = 1`

Why don't we have one for ioGetOutput?

**Reply by vurvdev** [reply]
You're free to make a PR to add it, just seems like a niche usecase
